packages/calypso-e2e/src/lib/components/domain-search-component.ts

```diff
@@ -74,13 +73,22 @@ export class DomainSearchComponent {
 	 * @returns {string} Domain that was selected.
 	 */
 	async selectDomain( keyword: string ): Promise< string > {
-		const targetItem = await this.page.waitForSelector( selectors.resultItem( keyword ) );
-		// Heading element inside a given result contains the full domain name string.
-		const selectedDomain = await targetItem
-			.waitForSelector( 'h3' )
-			.then( ( el ) => el.innerText() );
+		const target = this.page.getByRole( 'button' ).filter( { hasText: keyword } );
+		await target.waitFor();
 
-		await Promise.all( [ this.page.waitForNavigation(), targetItem.click() ] );
+		// The `heading` element represents the entire domain (including the tld).
+		const selectedDomain = await target.getByRole( 'heading' ).innerText();
+
+		await target.click();
+
+		// If multiple domain selections are enabled, the Continue button appears
+		// on the right hand sidebar.
+		// See: 21483-explat-experiment
+		// Note: this page object does not currently support multiple domain selection.
+		await Promise.race( [
+			this.clickButton( 'Continue' ),
+			this.page.waitForURL( /start\/plans/ ),
+		] );
```

Comment:
We are not yet planning to support selecting multiple domains during onboarding in E2Es, though it's definitely something to consider soon. 

That means, for now, we do not care about the type of domain selection, whether it be single or multi-domain.